### PR TITLE
support HTTPs

### PR DESCRIPTION
**Pull request by bodsch**
add https support for http  connections
add more vm.memory metrics

### CODE REVIEW

**Comment by mpetazzoni** [comment]
@bodsch Thanks for the PR. Could you make sure to keep your changes only about the HTTPS support? (and not add new metrics; feel free to open a distinct PR for that though!)

**Comment by bodsch** [comment]
okay, i split the PR.

File: jenkins.py [hunk 1]
```diff
@@ -28,10 +28,32 @@
...
+    'vm.memory.heap.used':
+        Metric('jenkins.node.vm.memory.heap.used', 'gauge'),
+    'vm.memory.non-heap.committed':
+        Metric('jenkins.node.vm.memory.non-heap.committed', 'gauge'),
+    'vm.memory.non-heap.max':
+        Metric('jenkins.node.vm.memory.non-heap.max', 'gauge'),
+    'vm.memory.non-heap.init':
+        Metric('jenkins.node.vm.memory.non-heap.init', 'gauge'),
+    'vm.memory.non-heap.usage':
+        Metric('jenkins.node.vm.memory.non-heap.usage', 'gauge'),
```
Comment (nishantagarwal1990) [inline jenkins.py]: The added metrics can be included using the EnhancedMetrics True and IncludeMetrics option in the configuration file.

File: jenkins.py [hunk 2]
```diff
@@ -104,8 +105,11 @@ def _api_call(url, opener, http_timeout):
     parsed_url = urlparse.urlparse(url)
     url = '{0}://{1}{2}'.format(parsed_url.scheme, parsed_url.netloc, urllib_quote(parsed_url.path))
     try:
+        ctx = ssl.create_default_context()
+        ctx.check_hostname = False
+        ctx.verify_mode = ssl.CERT_NONE
         urllib2.install_opener(opener)
-        resp = urllib2.urlopen(url, timeout=http_timeout)
+        resp = urllib2.urlopen(url, timeout=http_timeout, context=ctx)
```
Comment (nishantagarwal1990) [inline jenkins.py]: ctx is not supported in python 2.6 which is present in some of Amazon's Linux.

File: jenkins.py (hunk 1 above)
Comment (bodsch) [reply jenkins.py]: okay.
I just haven't found an example or documentation.

File: jenkins.py (hunk 2 above)
Comment (bodsch) [reply jenkins.py]: :)
this snippets are my **first** step in python ...

File: jenkins.py (hunk 1 above)
Comment (nishantagarwal1990) [reply jenkins.py]: You can take a look at the README. There is a sample present there for reference.

**Comment by nishantagarwal1990** [comment]
Closing PR as the issue gets fixed in #10

**Pull request closed without merging**